src/main/kotlin/snyk/common/SnykCachedResults.kt

```diff
@@ -0,0 +1,58 @@
+package snyk.common
+
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.project.Project
+import io.snyk.plugin.events.SnykScanListener
+import io.snyk.plugin.snykcode.SnykCodeResults
+import snyk.container.ContainerResult
+import snyk.iac.IacResult
+import snyk.oss.OssResult
+
+@Service
+class SnykCachedResults(val project: Project) {
+
+    var currentOssResults: OssResult? = null
+        get() = if (field?.isExpired() == false) field else null
+
+    var currentContainerResult: ContainerResult? = null
+        get() = if (field?.isExpired() == false) field else null
+
+    var currentIacResult: IacResult? = null
+        get() = if (field?.isExpired() == false) field else null
+
+    fun initCacheUpdater() = project.messageBus.connect().subscribe(
+        SnykScanListener.SNYK_SCAN_TOPIC,
+        object : SnykScanListener {
+
+            override fun scanningStarted() {}
```

Comment:
These empty code blocks can be removed, I guess?

Reply:
Unfortunately, not really :( we have to implement interface methods even if we don't need to do anything there. The replacement for `{}` could be ` = Unit` but it is even longer and imho less readable.